### PR TITLE
Magic

### LINKED ISSUE #10303: Request: magic
### Brand Name

magic

### Website

https://www.magic.link

### Popularity Metric

The SimilarWeb rank is 83,573. See https://www.similarweb.com/website/magic.link/#overview

### Official Resources for Icon and Color

https://www.magic.link

### Additional Comments

_No response_

### PR DESCRIPTION
**Pull request by durgeshgowdac**
**Issue:** closes #10303

**Website:** https://www.magic.link/
**Guidelines:** https://magic.link/brand-assets

**Similarweb rank:** 83,573
**Similarweb link:** https://www.similarweb.com/website/magic.link/#overview

### Checklist

- [x] I updated the JSON data in `_data/simple-icons.json`
- [x] I optimized the icon with SVGO or SVGOMG
- [x] The SVG `viewbox` is `0 0 24 24`

### CODE REVIEW

**Comment by durgeshgowdac** [comment]
@PeterShaggyNoble All checks have passed. You can merge now

File: _data/simple-icons.json [hunk 1]
```diff
@@ -8483,6 +8483,12 @@
             "hex": "EE672F",
             "source": "https://magento.com"
         },
+        {
+            "title": "Magic",
+            "hex": "6851FF",
+            "source": "https://magic.link",
```
Comment (PeterShaggyNoble) [inline _data/simple-icons.json:8489]:
```suggestion
            "source": "https://magic.link/brand-assets",
```

**Review by PeterShaggyNoble — CHANGES_REQUESTED** [review]
@durgeshgowdac, which file did you base your icon off? I'm seeing [significant differences](https://github.com/simple-icons/simple-icons/assets/15157491/85b2e18b-191e-4424-bcc7-e93d1c59755c) when comparing it (in red) to `\Magic Logo≡ƒ¬ä\SVG\01-Icon\2-Icon_Magic_Black.svg` from [the zip file](https://magic.link/brand-assets/magic-logos.zip) of logos they make available.

And we should update the `source` to match the `guidelines`.

**Comment by durgeshgowdac** [comment]
@PeterShaggyNoble I have made the changes, please check.

**Review by PeterShaggyNoble — APPROVED** [review]
Looks spot on now, @durgeshgowdac 👍🏻 Nice work.

**Pull request merged**